GH-22: Fix zwave_command_class_node_info_resolver unit test

silabs-borisl opened the pull request:
GH-22 add a new callback on the give_up_listener on the ATTRIBUTE_ZWAVE_SECURE_NIF attribute causing the test case to fail.

The test only expect this listener to be called on the non-secure NIF ATTRIBUTE_ZWAVE_NIF. This change make sure that the code is only executed in the context of ATTRIBUTE_ZWAVE_NIF and ignored  for ATTRIBUTE_ZWAVE_SECURE_NIF as this test don't care about it.

Bug-SiliconLabs: UIC-3082
Relate-to: https://github.com/SiliconLabs/UnifySDK/pull/22
Forwarded-SiliconLabs: task/UIC-3082/phcoval/GH-22/develop

## Change
<!--
  Describe your changes below.

  (internal references are encouraged in commit messages as well,
  please align to others changes)

-->

## Checklist
<!--
  Please put an `x` in each box to make sure to enable contribution process
-->

- [x] A [Contribution License Agreement][CLA] has been established between @SiliconLabs and author's company (matching email domain)

[CLA]: https://en.wikipedia.org/wiki/Contributor_License_Agreement

silabs-borisl committed:
GH-22: Fix zwave_command_class_node_info_resolver unit test

GH-22 add a new callback on the give_up_listener on the ATTRIBUTE_ZWAVE_SECURE_NIF attribute causing the test case to fail.

The test only expect this listener to be called on the non-secure NIF ATTRIBUTE_ZWAVE_NIF. This change make sure that the code is only executed in the context of ATTRIBUTE_ZWAVE_NIF and ignored  for ATTRIBUTE_ZWAVE_SECURE_NIF as this test don't care about it.

Bug-SiliconLabs: UIC-3082
Relate-to: https://github.com/SiliconLabs/UnifySDK/pull/22
Forwarded-SiliconLabs: task/UIC-3082/phcoval/GH-22/develop

rzr commented:
LGTM

Build: https://github.com/silabs-borisl/UnifySDK/actions/runs/7568335791/job/20609337322

rzr commented:
Please merge if you like the proposed change and it will land in upcoming release

rzr commented:
Feedback welcome, if no objection we will probably merge this downstream

rzr committed:
GH-22: Lint code for SonarCube

Bug-SiliconLabs: UIC-3082
Relate-to: https://github.com/SiliconLabs/UnifySDK/pull/22
Forwarded-SiliconLabs: task/UIC-3082/phcoval/GH-22/develop
Origin: https://github.com/silabs-borisl/UnifySDK/pull/2
Signed-off-by: Philippe Coval <philippe.coval@silabs.com>

silabs-borisl committed:
GH-22: Add test coverage for new function
